src/workerd/io/trace.h

```diff
@@ -609,6 +616,7 @@ class WorkerTracer final: public kj::Refcounted, public BaseTracer {
 
   void addLog(kj::Date timestamp, LogLevel logLevel, kj::String message) override;
   void addSpan(CompleteSpan&& span) override;
+  void setTraceId(tracing::TraceId traceId);
```

Comment:
Nit: `const tracing::TraceId&` to prevent the extra copy? I know it's a small class but still.